### ISSUE TITLE
[Bug]: Having a 2nd Virtual Display: Settings doesn't survive machine restart

### ISSUE THREAD

**Issue by SPYderHASS** [itsmikethetech/Virtual-Display-Driver#178]
### Which OS?

Win10 Pro

### Which release?

22H2

### Describe the bug

I have 1 physical display: 
In addition, I installed 2 Virtual Displays and setup them to different target resolutions/scalings. 1st observation: Changing scaling on one VD also effects the other VD (by did a scaling change too) - but it is not the same scaling percentage. I just want to mention that they effect each other. Apart from that, everything works as expected.

If I now shutdown/restart my entire machine, both VDs are having resolution set to 640x480 now. Also positioning of both is in sequence right next to my physical monitor on the top right. The entire settings behavior looks for me like some kind of "back to default".

If I now de-install one of the VDs using Device Manager, the now left-over VD getting the latest correct (aka wanted) settings automatically without any manual interaction. This 1 VDs works as expected now, also after several restarts.

### Steps to reproduce

- Install VD driver and Display adapter like described here for a 1st VD
- Install a 2nd VD following the steps you did for the 1st one.
- Setup both displays to the settings you want
- Proof that everything is fine and expected
- Restart your machine
- Open up Display Manager
- Find boths VDs set to resolution 640x480

### Expected behavior

Expect to have at least 2 VDs setup properly also after restart of the entire machine.

**Comment by bud3699** [comment]
The release was supposed to be whether the beta or not, I'll update that when I'm home, just at hospital right now, if you havnt already can you try the beta and enable logs, reproduce the issue then send it

**Comment by bud3699** [comment]
Wait did you add the driver multiple times for multiple displays, you change the display count in option.txt or vdd_setting.xml

**Comment by SPYderHASS** [comment]
Hi bud3699,

first of all...please stay focused on your health. There is nothing more important than that!

tl;dr: Within the beta of VDD - Increase monitor count to "2" resolved the issue.

Thanks to your quick reply. I've tried a combination of what you suggested. As I wasn't aware of file "options.txt" I spend looking to it. As it is a "not so speaking" text file, I moved forward to download and install the latest beta of VDD as you suggested. As I now have a more speaking xml-structure here (vdd_setting.xml) I found the monitor counter and increased to "2". Afterwards I installed the virtual driver ending up with 2 displays directly. I edited "vdd_setting.xml" a bit more and entered my target resolution as it wasn't there (SteamDeck Super Sampling: 1920x1200) I was able to setup every VD as I wanted. This setup now survived 2 machine restarts now. I would say there is no issue anymore. This ticket can be closed.

I would leave it up to you to close the ticket as you might to comment something more? If you can't close for any reason, I will close the ticket as "resolved" in the next couple of days.

A lot of thanks to you and the entire team for your incredible work and support. All the best for you!

**Comment by bud3699** [comment]
> Hi bud3699,
> 
> first of all...please stay focused on your health. There is nothing more important than that!

I'm A-okay! Just got home, Nothing but a couple cuts and bruises everywhere, I'm fine! thank you for your concern, I have my pain killers, aha.



> Thanks for your quick reply. I've tried a combination of what you suggested. As I wasn't aware of file "options.txt" I spend looking to it. As it is a "not so speaking" text file, I moved forward to download and install the latest beta of VDD as you suggested. As I now have a more speaking xml-structure here (vdd_setting.xml) I found the monitor counter and increased to "2". Afterwards I installed the virtual driver ending up with 2 displays directly. I edited "vdd_setting.xml" a bit more and entered my target resolution as it wasn't there (SteamDeck Super Sampling: 1920x1200) I was able to setup every VD as I wanted. This setup now survived 2 machine restarts now. I would say there is no issue anymore. This ticket can be closed.

You're very welcome for the help, glad you got it set up, I'll close the ticket


> I would leave it up to you to close the ticket as you might to comment something more? If you can't close for any reason, I will close the ticket as "resolved" in the next couple of days.

I shall close the ticket, if you need any help or have any more questions feel free to join the [discord](http://discord.mikethetech.com/) 


> A lot of thanks to you and the entire team (aka Mike :-) ) for your incredible work and support. All the best for you!

Thank you for the appreciation, Progress has been sky rocketing and we aim to be pushing out a new beta soon, I'm just finishing up a couple of things to try juice up the update!
